### PR TITLE
Small new features and fixes

### PR DESCRIPTION
**Pull request by anaston**
I added 2-2 small features and fixes.

Added features:

- Conn.threshold method for lesioning network
- seed to Conn.add_weights for reproducibility

Fixes:

- np.bool update for numpy 1.26.2 compatibility
- update the default value in performance.balanced_accuracy_score (from adjusted=None to True, note that in sklearn.metrics.balanced_accuracy_score the default value is False)

### CODE REVIEW

**Review by estefanysuarez — APPROVED** [review]
Thank you Agoston!
